src/west/commands/project.py

```diff
@@ -946,19 +855,73 @@ def _update_west():
             raise WestUpdated()
 
 
-def _update_and_reset_west():
-    # Updates west by resetting to the new revision after fetching it
-    # (with 'git reset --keep').
+def _update(project, rebase, keep_descendants):
+    _fetch(project)
 
-    project = _west_project()
-    with _error_context(', while updating/resetting west'):
+    branch = _current_branch(project)
+    sha = _sha(project, _MANIFEST_REV_BRANCH)
+    if branch is not None:
+        is_ancestor = _is_ancestor_of(project, sha, branch)
+        try_rebase = rebase
+    else:
+        # If no branch is checked out, -k and -r don't matter.
```

Comment:
Probably want to update the comments in this function to not use -k or -r, since rebase and keep_descendants are formal parameters rather than command line arguments from this function's perspective.

Reply:
fixed

Reply:
There were two comments to update; the other one is:

```
        # A descendant is currently checked out and -k was
        # given, so there's nothing more to do.
```

Reply:
fixed